cmd/todd-agent/main.go

```diff
@@ -51,6 +51,11 @@ func main() {
 
 	cfg := config.GetConfig(arg_config)
 
+	// Create optDir
+	if err := os.MkdirAll(cfg.LocalResources.OptDir, 0755); err != nil {
```

Comment:
I agree. I think it depends on how you intend for todd to be distributed/installed. My thinking on moving it into the binary was that the binary alone would enough. Rather than requiring the Makefile. I also imagine there are a number of folks that don't have make installed.

Getting the permission error gives the user the choice to either create the directory themselves or run todd with elevated permissions. Admittedly, a message to that affect would be a better user experience.